test/hotspot/jtreg/serviceability/attach/LongArgTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Tests that long arguments of attach operation are not truncated
+ * @bug 8334168
+ * @library /test/lib
+ * @modules jdk.attach/sun.tools.attach
+ * @run main LongArgTest
+ */
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.io.IOException;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.tools.attach.HotSpotVirtualMachine;
+
+import jdk.test.lib.apps.LingeredApp;
+
+public class LongArgTest {
+
+    public static void main(String[] args) throws Exception {
+        LingeredApp app = null;
+        try {
+            app = LingeredApp.startApp();
+
+            test(app).run();
+
+            test(app)
+                .withLongValue()
+                .run();
+
+            test(app)
+                .withSuperLongValue()
+                .run();
+
+        } finally {
+            LingeredApp.stopApp(app);
+        }
+
+    }
+
+    // For simplicity, the test uses internal HotSpotVirtualMachine,
+    // sets/gets "HeapDumpPath" flag value (string flag, not validated by JVM).
+    private static class Test {
+        private LingeredApp app;
+        private String flagName = "HeapDumpPath";
+        private String flagValue = generateValue(5);
+
+        Test(LingeredApp app) {
+            this.app = app;
+        }
+
+        // Value length exceeds 1K.
+        Test withLongValue() {
+            flagValue = generateValue(1024 + 1);
```

Comment:
Shouldn't we also be testing exactly 1024 and expect it to work?

Reply:
good question!
tried to make it 1024 and it does not work!
Windows VirtualMachineImpl has one more issue - it uses 1024 char buffers, but it should be (1024 + 1) as AttachListener does (1 extra char to 0-terminator), so maximum argument size on windows is 1023.
Will fix it.

Reply:
Fixed the issue, also renamed ` jstring_to_cstring` argument to make it clear that buffer size is expected and updated callers to use `sizeof`.